sv.h

```diff
@@ -1859,19 +1859,30 @@ scalar.
 #define SvPV_const(sv, len)   SvPV_flags_const(sv, len, SV_GMAGIC)
 #define SvPV_mutable(sv, len) SvPV_flags_mutable(sv, len, SV_GMAGIC)
 
-#define SvPV_flags(sv, len, flags) \
-    (SvPOK_nog(sv) \
+/* This test is "is there a cached PV that we can use directly?"
+ * We can if
+ * a) SVf_POK is true and there's definitely no get magic on the scalar
+ * b) SVp_POK is true, there's no get magic, and we know that the cached PV
+ *    came from an IV conversion.
+ * For the latter case, we don't set SVf_POK so that we can distinguish whether
+ * the value originated as a string or as an integer, before we cached the
+ * second representation. */
+#define SvPOK_or_cached_IV(sv) \
+    (((SvFLAGS(sv) & (SVf_POK|SVs_GMG)) == SVf_POK) || ((SvFLAGS(sv) & (SVf_IOK|SVp_POK|SVs_GMG)) == (SVf_IOK|SVp_POK)))
```

Comment:
In what cases testing for just `SVp_POK|SVs_GM == SVp_POK` wouldn't be enough?

I always thought that `SVp_[INP]OK` flags were supposed to mean "this scalar has a string/numeric representation" while `SV_[INP]OK` mean "this scalar has a **canonical/lossless** string/numeric representation". If my understanding is correct, SvPV should have been testing for `SVp_POK` instead of `SVf_POK` even without the changes introduced in this PR.

Reply:
`SvPV()` has been testing only `SVf_POK` for a very long time, since 463ee0b2acb in 1993, if we do change it, I don't think it belongs in this PR.